run.py

```diff
@@ -1,261 +1,51 @@
-        "projects": export_projects,
-        "certifications": st.session_state.certifications.split('\n'),
-        "skills": export_skills
-    }
-    st.session_state['output_json'] = export
-    os.makedirs('output', exist_ok=True)
-    with open('./output/export_resume.json', 'w', encoding='utf-8') as f:
-        json.dump(export, f, ensure_ascii=False, indent=4)
-    st.toast("Form submitted!", icon="🎯")
-
-def downloader_callback():
-    if st.session_state['output_json'] is None:
-        st.toast(":red[Submit the form first!]", icon="⚠️")
-        return
-    st.toast("New resume downloaded!", icon="🎯")
-    with open('./output/export_resume_doc.json', 'w', encoding='utf-8') as f:
-        json.dump(st.session_state['output_json'], f, ensure_ascii=False, indent=4)
-
-def uploader_callback():
-    st.toast("Resume uploaded.", icon="📑")
-    shutil.rmtree("./tmp_output", True )
-
-    st.session_state['parsed_pdf'] = dict()
-    st.session_state['processed'] = False
-    st.session_state['output_json'] = None
-
-##################################### INITIALIZE STATES ################################################
-def init_state(key, value):
-    if key not in st.session_state:
-        st.session_state[key] = value
-
-init_state('parsed_pdf', dict())
+import streamlit as st
+
+from src.parsing.process_cv import parsing_cv
+from src.helpers.utils import init_state, display_pdf_pil
+from src.helpers.callbacks import uploader_callback, downloader_callback
+from src.parsing.post_process import (write_description, reset_description, 
+                                  rewrite_resp, reset_resp, infer_more_skills, submit_form)
+
+init_state('parsed_pdf', {})
 init_state('uploaded', False)
 init_state('processed', False)
 init_state('output_json', None)
+
 ##################################### LAYOUT DEFINITION ################################################
 st.set_page_config(page_title="Resume Parser", page_icon="📑")
-st.title("📑 Resume Parser",)
-
-# Inject custom CSS to set the width of the sidebar
-st.markdown(
-    """
-    <style>
-        section[data-testid="stSidebar"] {
-            width: 100% !important; # Set the width to your desired value
-        }
-    </style>
-    """,
-    unsafe_allow_html=True,
-)
-
-# st.button("Restart", on_click="restart")
+st.title("📑 Resume Parser")
+
 with st.sidebar:
     # resume upload
     uploaded_file = st.file_uploader("Upload Resume", on_change=uploader_callback)
     if uploaded_file is not None:       
-        display_pdf(uploaded_file=uploaded_file)
+        display_pdf_pil(uploaded_file=uploaded_file)
 
 if st.session_state['processed']:
     status = st.status("Editing the resume ... ", expanded=True)
 elif (uploaded_file is not None):
     status = st.status("Processing the resume ... ", expanded=True) 
 
 if (uploaded_file is not None) and (not(st.session_state['processed'])):
-    # extract_text_from_pdf(uploaded_file=uploaded_file)
-    status.write("📝 Extracting text...")
```

Comment:
Why remove?

Reply:
status.write("📝 Extracting text...")

Reply:
pls check file process.cv, it should be inside there